meta-balena-common/recipes-core/initrdscripts/files/fsck

```diff
@@ -24,4 +24,14 @@ fsck_run() {
     fsck.ext4 -p /dev/disk/by-label/resin-rootA
     fsck.ext4 -p /dev/disk/by-label/resin-rootB
     fsck.ext4 -p /dev/disk/by-label/resin-state
+
+    # Correct GPT secondary header
+    datapart=$(readlink -f /dev/disk/by-label/resin-data)
+    datadev=$(lsblk $datapart -n -o PKNAME)
+    if [ "gpt" = "$(blkid -o value -s PTTYPE /dev/$datadev)" ] &&
+            ( sgdisk --verify /dev/$datadev | grep -q "[T|t]he secondary header" ) ; then
+        msg "Moving secondary GPT header to the end of the disk."
+        sgdisk --move-second-header /dev/$datadev
+        partprobe
+    fi
 }
```

Comment:
This is already done at https://github.com/balena-os/meta-balena/blob/26f67abe90faf461ba9352bc5b7627e6a250165e/meta-balena-common/recipes-core/initrdscripts/files/resindataexpander#L27